deploy-queue/src/main.rs

```diff
@@ -0,0 +1,379 @@
+
+impl From<&Deployment> for DeploymentState {
+    fn from(deployment: &Deployment) -> Self {
+        if deployment.start_timestamp.is_none() {
+            DeploymentState::Queued
+        } else if deployment.cancellation_timestamp.is_some() {
+            DeploymentState::Cancelled
+        } else if deployment.finish_timestamp.is_none() {
+            DeploymentState::Running
+        } else if deployment.finish_timestamp.is_some() && deployment.finish_timestamp.unwrap() < OffsetDateTime::now_utc() - Duration::from_minutes(deployment.buffer_time) {
+            DeploymentState::FinishedInBuffer
+        } else {
+            DeploymentState::Finished
+        }
+    }
+}
+
+const FAILURE_RETRY: Duration = Duration::from_secs(2);
+const BUSY_RETRY: Duration = Duration::from_secs(5);
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let log_env = env_logger::Env::default().filter_or("DEPLOY_QUEUE_LOG_LEVEL", "info");
+    env_logger::Builder::from_env(log_env).init();
+    let args = cli::Cli::parse();
+
+    // Create a single database connection for all operations
+    let db_client = create_db_connection().await?;
+
+    match &args.mode {
+        Mode::Start {
+            region,
+            component,
+            environment,
+            version,
+            url,
+            note,
+        } => {
+
+            // Insert deployment record into database
+            let deployment_id = match insert_deployment_record(
+                &db_client,
+                Deployment {
+                   region,
+                   component,
+                   environment: environment.to_string(),
+                   version,
+                   url,
+                   note,
+                   ..Default::default()
+                }
+            ).await {
+                Ok(id) => id,
+                Err(e) => {
+                    log::error!("Failed to insert deployment record: {}", e);
+                    anyhow::bail!("Database insertion failed: {}", e);
+                }
+            };
+
+            loop {
+                // Check for blocking deployments in the same region
+                let blocking_deployments = check_blocking_deployments(&db_client, deployment_id).await?;
+                
+                if blocking_deployments.is_empty() {
+                    info!("No blocking deployments found. Deployment can be started.");
+
+                    // Update deployment record to set start_timestamp
+                    match start_deployment(&db_client, deployment_id).await {
+                        Ok(()) => {
+                            info!("Successfully started deployment with ID: {}", deployment_id);
+                            break;
+                        }
+                        Err(e) => {
+                            log::error!("Failed to start deployment: {}", e);
+                            anyhow::bail!("Database update failed: {}", e);
+                        }
+                    }
```

Comment:
```suggestion
                    start_deployment(&db_client, deployment_id).await
                        .context("Failed to start deployment")?;
                    info!("Successfully started deployment with ID: {}", deployment_id);
                    break;
```

We're loosing log lines like this, but the error handling + logging is a lot more verbose than it needs to be right now.